defi/src/adaptors/handlers/storeAdaptorData/index.ts

```diff
@@ -206,19 +212,49 @@ export const handler2 = async (event: IStoreAdaptorDataHandlerEvent) => {
       } else
         throw new Error("Invalid adapter")
 
+
+      if (runType === 'store-all') {
+        let runAtCurrTime = adaptersToRun.some(([_version, adapter]) => Object.values(adapter).some(a => a.runAtCurrTime))
+
+        const date = new Date()
+        const hours = date.getUTCHours()
+        const isExpensiveAdapter = adaptor.isExpensiveAdapter
+        // if it is an expensive adapter run every 4 hours or after 20:00 UTC
+        const runNow = !isExpensiveAdapter || (hours % 4 === 0 || hours > 20)
+        const haveTodayData = todayIdSet.has(id)
+        const haveYesterdayData = yesterdayIdSet.has(id)
+
+
+        if (runAtCurrTime || !isAdapterVersionV1) {
+          recordTimestamp = timestampAtStartofHour
+          endTimestamp = timestampAtStartofHour
+
+          if (haveTodayData && !runNow) {
+            console.info(`Skipping ${adapterType} - ${index + 1}/${protocols.length} - ${protocol.module} - skipping, already have today data for adapter running at current time`)
+            return
+          }
+        } else { // it is a version 1 adapter - we pull yesterday's data
+          if (haveYesterdayData) {
+            console.info(`Skipping ${adapterType} - ${index + 1}/${protocols.length} - ${protocol.module} - skipping, already have yesterday data`)
+            return;
+          }
+
+          endTimestamp = getTimestampAtStartOfDayUTC(Math.floor(Date.now() / 1000) - 1)
+          recordTimestamp = getTimestampAtStartOfDayUTC(endTimestamp)
```

Comment:
why use getTimestampAtStartOfDayUTC() here?

Reply:
turns 12-12-2025T23:59:59 into 12-12-2025T00:00:00 
since we use from timestamp as record timestamp for v1 adapters as I have mentioned earlier

Reply:
yeah but endTimestamp should already be at 0:00?

Reply:
good catch, that -1 line above must be one level above